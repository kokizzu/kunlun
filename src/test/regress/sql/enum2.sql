--DDL_STATEMENT_BEGIN--
create type vihicles as enum ('car', 'suv', 'truck', 'mpv', 'wagon', 'hatchback');
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
drop table if exists tvehicles;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
create table tvehicles (a int, b vihicles);
--DDL_STATEMENT_END--
insert into tvehicles values(1, 'car'),(2, 'suv'),(3, 'truck'),(4, 'hatchback'),(5, 'mpv'), (6, 'wagon');
select*from tvehicles;
select*from tvehicles where b > 'suv';
select*from tvehicles where b <= 'mpv';
select*from tvehicles where a>2;
select*from tvehicles where b='suv';
update tvehicles set b='mpv' where b='truck';
select*from tvehicles order by a;
delete from tvehicles where b='mpv';
select*from tvehicles;
delete from tvehicles where b='suv';
select*from tvehicles;
update tvehicles set b='mpv';
select*from tvehicles;

