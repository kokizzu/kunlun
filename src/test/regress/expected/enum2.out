create type vihicles as enum ('car', 'suv', 'truck', 'mpv', 'wagon', 'hatchback');
CREATE TYPE
drop table if exists tvehicles;
psql:sql/enum2.sql:2: NOTICE:  table "tvehicles" does not exist, skipping
DROP TABLE
create table tvehicles (a int, b vihicles);
CREATE TABLE
insert into tvehicles values(1, 'car'),(2, 'suv'),(3, 'truck'),(4, 'hatchback'),(5, 'mpv'), (6, 'wagon');
INSERT 0 6
select*from tvehicles;
 a |     b     
---+-----------
 1 | car
 2 | suv
 3 | truck
 4 | hatchback
 5 | mpv
 6 | wagon
(6 rows)

select*from tvehicles where b > 'suv';
 a |     b     
---+-----------
 3 | truck
 4 | hatchback
 5 | mpv
 6 | wagon
(4 rows)

select*from tvehicles where b <= 'mpv';
 a |   b   
---+-------
 1 | car
 2 | suv
 3 | truck
 5 | mpv
(4 rows)

select*from tvehicles where a>2;
 a |     b     
---+-----------
 3 | truck
 4 | hatchback
 5 | mpv
 6 | wagon
(4 rows)

select*from tvehicles where b='suv';
 a |  b  
---+-----
 2 | suv
(1 row)

update tvehicles set b='mpv' where b='truck';
UPDATE 1
select*from tvehicles order by a;
 a |     b     
---+-----------
 1 | car
 2 | suv
 3 | mpv
 4 | hatchback
 5 | mpv
 6 | wagon
(6 rows)

delete from tvehicles where b='mpv';
DELETE 2
select*from tvehicles;
 a |     b     
---+-----------
 1 | car
 2 | suv
 4 | hatchback
 6 | wagon
(4 rows)

delete from tvehicles where b='suv';
DELETE 1
select*from tvehicles;
 a |     b     
---+-----------
 1 | car
 4 | hatchback
 6 | wagon
(3 rows)

update tvehicles set b='mpv';
UPDATE 3
select*from tvehicles;
 a |  b  
---+-----
 1 | mpv
 4 | mpv
 6 | mpv
(3 rows)

